src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (isEmpty()) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        size++;
+        tail = node;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index
+                    + " out of range " + size);
+        } else if (index == 0 && head != null) {
+            Node<T> newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+        } else if (index == size) {
+            add(value);
+        } else {
+            Node<T> oldNode = getNode(index);
+            Node<T> newNode = new Node<>(oldNode.prev, value, oldNode);
+            oldNode.prev.next = newNode;
+            oldNode.prev = newNode;
+            size++;
+        }
```

Comment:
Algorithm: 
1. if (index == size) call method add(T value) and return. 
2. check index 
3. get nextNode by index
4. prevNode = nextNode.prev;
5. newNode = new Node<>(prev, value, next);
6. if prevNode == null than head = newNode 
7. else prevNode.next = newNode
8. after if-else: nextNode.prev = newNode and increase size